pipeline/transformer/transformer_sierra/src/main/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/SierraItemsOnOrder.scala

```diff
@@ -0,0 +1,182 @@
+package uk.ac.wellcome.platform.transformer.sierra.transformers
+
+import grizzled.slf4j.Logging
+import uk.ac.wellcome.platform.transformer.sierra.source.{
+  SierraItemData,
+  SierraOrderData
+}
+import weco.catalogue.internal_model.identifiers.IdState
+import weco.catalogue.internal_model.locations.{LocationType, PhysicalLocation}
+import weco.catalogue.internal_model.work.Item
+import weco.catalogue.source_model.sierra.{
+  SierraItemNumber,
+  SierraOrderNumber,
+  TypedSierraRecordNumber
+}
+
+import java.text.SimpleDateFormat
+import java.util.Date
+import scala.util.Try
+
+/** This transformer creates catalogue items that correspond to items that
+  * are "on order" or "awaiting cataloguing" -- which don't have their own
+  * item record yet.
+  *
+  *
+  * To understand how this works, it's useful to understand the ordering
+  * process:
+  *
+  *   1)  A staff member orders a book.  They create a skeleton bib record
+  *       in Sierra, which is linked to an order record with status 'o' ("on order").
+  *       At this point, there are no item records.
+  *
+  *   2)  When the book arrives, it's "received".  The RDATE on the order
+  *       record gets populated, the status is updated to 'a' ("fully paid"),
+  *       and invoice information gets attached to the order record.
+  *
+  *   3)  At some point after that, an item record is created.  This supercedes
+  *       the order record.
+  *
+  * The Sierra documentation for fixed fields on order records is useful reading:
+  * https://documentation.iii.com/sierrahelp/Default.htm#sril/sril_records_fixed_field_types_order.html%3FTocPath%3DSierra%2520Reference%7CHow%2520Innovative%2520Systems%2520Store%2520Information%7CFixed-length%2520Fields%7C_____11
+  *
+  */
+object SierraItemsOnOrder extends Logging {
+  def apply(
+    id: TypedSierraRecordNumber,
+    itemDataMap: Map[SierraItemNumber, SierraItemData],
+    orderDataMap: Map[SierraOrderNumber, SierraOrderData]
+  ): List[Item[IdState.Unidentifiable.type]] =
+    if (itemDataMap.isEmpty) {
```

Comment:
I wonder if it would be clearer for this conditional to live at https://github.com/wellcomecollection/catalogue/pull/1568/files#diff-9f82908b0b203caf49a1bf7baed59891b25af2898e2ecbf6b215f1fa57c4f916R118 so we don't have to pass `itemDataMap` so opaquely when we don't really use it

Reply:
I've swapped it out for a `hasItems` boolean.